### DISCUSSION TITLE
Nginx location /mail

### DISCUSSION THREAD

**Discussion by Skulaurun** [postalserver/postal#2100] [Help with using Postal] [answered]
Hello,

I would like to run the postal web interface as a subdirectory (`/mail`), not as a subdomain/domain.

This is my nginx configuration to achieve this:
```
location /mail {
        proxy_pass http://127.0.0.1:5000/;
        proxy_http_version 1.1;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
}
```

However when I try accessing the web interface on `https://www.mydomain.com/mail`, the URL becomes `https://www.mydomain.com/login?return_to=%2F`, thus removing the `/mail` part, as it is expecting to be run in the domain root (`/`).

I don't know how to solve this via nginx configuration, and I think this is something to be configured in the web interface source. For example using the `<base>` html tag.

So how do I make this work with postal? So the web page expects to be run in `/mail` directory?

Thanks in advance.

**Comment by willpower232** [comment, marked as answer]
I think you'd have to modify the [routes file](https://github.com/postalserver/postal/blob/main/config/routes.rb) to include a subfolder like this https://the0x00.dev/ruby/How-To-Run-Rails-In-Subfolder/
